Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/AnalyticsSettings.kt

```diff
@@ -1,7 +1,35 @@
 package de.rki.coronawarnapp.datadonation.analytics
 
+import android.content.Context
+import com.google.protobuf.ByteString
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData.ExposureRiskMetadata
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.FlowPreference
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class AnalyticsSettings @Inject constructor()
+class AnalyticsSettings @Inject constructor(
+    @AppContext private val context: Context
+) {
+    private val prefs by lazy {
+        context.getSharedPreferences("analytics_localdata", Context.MODE_PRIVATE)
+    }
+
+    val previousExposureRiskMetadata = FlowPreference(
+        preferences = prefs,
+        key = PREVIOUS_EXPOSURE_RISK_METADATA,
+        reader = { key ->
+            getString(key, null)?.let {
+                ExposureRiskMetadata.parseFrom(ByteString.copyFromUtf8(it))
+            }
+        },
+        writer = { key, value ->
+            putString(key, value?.toByteString()?.toStringUtf8())
+        }
+    )
```

Comment:
Be cautious here, we are using two different implementations for data conversion.

* On read: we use `ByteString` from `com.google.protobuf`
* On write: we use `ByteString` conversion functions from Kotlin

Can we use read/write from the same implementation source?
e.g. for a Kotlin only solution:

```kotlin
        ByteArray(1).toByteString().base64()
        "".decodeBase64()!!.toByteArray()
```

Reply:
done 👍